### PR TITLE
Support for ElfinH7 Wing

### PR DESCRIPTION
**Pull request by suiaiyun**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/ElfinH7-Wing/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+ - 7x UARTs (1,2,3,4,6,7,8)
+ - 13x PWM Outputs (12 Motor Output, 1 LED)
+ - 1x RC input PWM/PPM, SBUS
+ - 2x I2C port for external compass, airspeed sensor, etc.
+ - SPI4 port
+ - USB port (with remote cabling)
+ - CAN port
+ - External Buzzer
+ - Dual Switchable Camera inputs
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/ElfinH7-Wing/README.md:20]: should mention how to control them lower in readme

File: libraries/AP_HAL_ChibiOS/hwdef/ElfinH7-Wing/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ - Built-in OSD
+- microSD card
+- Battery input voltage: 2S-6S
+ - BEC: 1x 5V/2A, 1x 9/12V/2A, 1x 5/6/7.2V/12A
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/ElfinH7-Wing/README.md:24]: should mention how to control voltage selection and power control down lower in readme

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
Some changes/expansions
also, since this would need a custom bootloader, that you will need to program on the device before shipping and instruction for users how to take the Ardupilot firmware and pack it for loading onto the SD card for your bootloader to push into SPI for execution,  we would not want to include an ArduPilot bootloader that would produce code that could overwrite it, so those files need to be removed

**Comment by suiaiyun** [comment]
> I also suggest you send early prototypes to a few ArduPilot devs so that they can verify and help ...Andrew Tridgell, Andy Piper at least...and I would like to have one to verify user instructions on firmware loading as I make the wiki page for the device...

Hello, do you need to mail it to the developers? May I ask where the address is? At present, this board only has a prototype board and has not started production. If necessary, I can mail the prototype board to the developers.

> also a link to your website for the device when its ready

There is currently no relevant website available. If necessary, I will create a corresponding website.

File: libraries/AP_HAL_ChibiOS/hwdef/ElfinH7-Wing/README.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ - PWM 5-8     in group2
+ - PWM 9,10   in group3
+ - PWM 11,12 in group4
+ - PWM 13      in group5
+
+Channels within the same group need to use the same output rate. If any channel in a group uses DShot then all channels in the group need to use DShot. Channels 1-4 support bi-directional dshot.
+
+## Battery Monitoring
+
+The board has a builting voltage and current sensor. The current sensor can read up to 130 Amps. The voltage sensor can handle up to 6S LiPo batteries.
```
[the remark this answers is not in the document]
Comment (suiaiyun) [reply libraries/AP_HAL_ChibiOS/hwdef/ElfinH7-Wing/README.md]: Correction made

**Comment by suiaiyun** [comment]
> There is no requirement to send the boards, but it will allow us to provide assistance during your development, help assure nothing is overlooked and provide a means to help answer questions on any issues that might arise from you or your customers more quickly.

This is great. Do I need to provide the schematic diagram of the board to the developers? As this is the first time submitting a PR for a new board, please forgive me for not clarifying what needs to be sent to the developers.

**Comment by Hwurzburg** [comment]
> > There is no requirement to send the boards, but it will allow us to provide assistance during your development, help assure nothing is overlooked and provide a means to help answer questions on any issues that might arise from you or your customers more quickly.
> 
> This is great. Do I need to provide the schematic diagram of the board to the developers? As this is the first time submitting a PR for a new board, please forgive me for not clarifying what needs to be sent to the developers.

having a schematic is always helpful....we are very careful to use only with lead developers...
I have discussed internally and the plane lead feels it important that I get a board as soon as possible to run user oriented trials on the external QSPI setup....its new and only one dev has played with this function....if that is possible....my email is hwurzburg@yahoo.com and I can send you details if you wish

**Comment by suiaiyun** [comment]
> having a schematic is always helpful....we are very careful to use only with lead developers... I have discussed internally and the plane lead feels it important that I get a board as soon as possible to run user oriented trials on the external QSPI setup....its new and only one dev has played with this function....if that is possible....my email is [hwurzburg@yahoo.com](mailto:hwurzburg@yahoo.com) and I can send you details if you wish

Hello. The schematic diagram of the ElfinH7 Wing board has been sent to your email. Thank you for your review.

**Comment by Hwurzburg** [comment]
@suiaiyun  any updates? is the project still alive?

**Comment by suiaiyun** [comment]
> @suiaiyun any updates? is the project still alive?

Yes, it is still alive, but encountered a problem. I asked a question in the discussion, but did not receive a clear response. https://discuss.ardupilot.org/t/plane-4-4-release/99954/141

**Comment by Hwurzburg** [comment]
@suiaiyun any update?

**Comment by Hwurzburg** [comment]
@peterbarker I suggest removing the board support label so it stops coming up on my list....its been abondoned or is inactive

**Comment by peterbarker** [comment]
@suiaiyun please open a fresh PR if you want to add support for this board; a new board ID will need to be allocated.

I strongly suggest you separate any other code changes from adding support for this new board - if you want to make logical changes to flight code that deserves separate PR, and I think that's what might have gone wrong with this one.

**Comment by Hwurzburg** [comment]
@suiaiyun  and I cant seem to find the email in August where you said you sent me the schematic...if you decide to reopen a PR for this board please resend the schematic...thanks

**Pull request closed without merging**